### PR TITLE
Year selectors UX

### PR DESCRIPTION
**Pull request by JoaquinTrinanes**
### General description

This PR changes the styles of the year selectors to fix the usability issues associated with them

### Designs

https://www.figma.com/file/hazwr4jct7j9Zeu99H3foJ/LG-Tool---Design-system?node-id=3761%3A4771

## Checklist before merging

- [x] Branch name / PR includes the related Jira ticket Id.
- [ ] Tests to check core implementation / bug fix added.
- [x] All checks in Continuous Integration workflow pass.
- [ ] Feature functionally tested by reviewer(s).
- [ ] Code reviewed by reviewer(s).
- [x] Documentation updated (README, CHANGELOG...) (if required)

### CODE REVIEW

File: client/package.json [hunk 1]
```diff
@@ -26,7 +26,7 @@
     "@dnd-kit/modifiers": "^5.0.0",
     "@dnd-kit/sortable": "^6.0.1",
     "@floating-ui/react-dom": "^0.7.0",
-    "@floating-ui/react-dom-interactions": "^0.6.3",
+    "@floating-ui/react-dom-interactions": "^0.6.4",
```
Comment (davidsingal) [inline client/package.json:29]: Used fixed versions please

**Comment by JoaquinTrinanes** [comment]
> A green border (focus) is not removable despite you clicking outside. It's only removed if you open and close the dropdown.

This is `react-select`'s default behaviour, I'll try to see if I can tweak it

> Also, ample space is not needed. In the design, the select is adapted to the content width. Screenshot 2022-06-27 at 11 16 02

With less width there's no space to show the `Type any year`, unless we increase it when it's open&hellip;

> Shouldn't be the blinking cursor at the beginning of the input?? Screenshot 2022-06-27 at 11 16 43

Yeah it should, `react-select` has a placeholder AND an input, in that order. It doesn't provide a way to change the order, only specific components, but I can try things like reversing the children order (so the input is first) or trying to add the placeholder to the input instead of another specific element.

File: client/src/components/select/component.tsx [hunk 2]
```diff
@@ -176,19 +171,45 @@ const Select: React.FC<SelectProps> = ({
...
+    ({ children, onChange, ...rest }) => {
+      return (
+        <components.Input
+          {...rest}
+          onChange={(e) => {
+            if (numeric) {
+              if (!/^[0-9]*$/.test(e.currentTarget.value)) {
+                return;
+              }
+            }
```
Comment (davidsingal) [inline client/src/components/select/component.tsx:184]:
```suggestion
            if (numeric && !/^[0-9]*$/.test(e.currentTarget.value)) return;
```

File: client/package.json [hunk 3]
```diff
@@ -59,7 +59,7 @@
     "@visx/tooltip": "2.1.0",
     "@visx/xychart": "2.9.0",
     "autoprefixer": "10.2.5",
-    "axios": "0.21.1",
+    "axios": "^0.27.2",
```
Comment (davidsingal) [inline client/package.json:62]: Same as above, fixed versions

**Review by davidsingal — CHANGES_REQUESTED** [review]

**Comment by davidsingal** [comment]
Screenshot 2022-06-27 at 11 45 10
Maybe, react-select put the placeholder over the input.

**Review by davidsingal — APPROVED** [review]

**Pull request merged**